WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageParentViewModel.kt

```diff
@@ -128,4 +152,74 @@ class PageParentViewModel
         }
         return children + grandchildren
     }
+
+    fun onSearchExpanded(restorePreviousSearch: Boolean) {
+        if (isSearchExpanded.value != true) {
+            if (!restorePreviousSearch) {
+                clearSearch()
+            }
+            _isSearchExpanded.value = true
+        }
+    }
+
+    fun onSearchCollapsed() {
+        _isSearchExpanded.value = false
+        clearSearch()
+
+        launch {
+            delay(SEARCH_COLLAPSE_DELAY)
+        }
+    }
+
+    private fun clearSearch() {
+        _lastSearchQuery = ""
+        _searchPages.postValue(null)
+        _isSaveButtonVisible.postOnUi(false)
+        _currentParent.isSelected = false
+    }
+
+    fun onSearch(searchQuery: String, delay: Long = SEARCH_DELAY) {
+        searchJob?.cancel()
+        if (searchQuery.isNotEmpty()) {
+            searchJob = launch {
+                delay(delay)
+                searchJob = null
+                if (isActive) {
+                    _lastSearchQuery = searchQuery
+                    val result = search(searchQuery)
+                    _searchPages.postValue(result)
+                }
+            }
+        } else {
+            clearSearch()
+        }
+    }
+
+    private suspend fun search(
+        searchQuery: String
+    ): List<PageItem> = withContext(defaultDispatcher) {
+        val tempPages = _pages.value
+        if (tempPages != null) {
+            if (tempPages.isNotEmpty()) {
+                return@withContext tempPages
+                        .filterIsInstance(ParentPage::class.java)
+                        .filter { it.id != 0L }
+                        .filter {
+                            it.title.toLowerCase(Locale.ROOT).contains(
+                                    searchQuery.toLowerCase(
+                                            Locale.ROOT
+                                    )
+                            )
+                        }
+            }
+        }
+        return@withContext mutableListOf<PageItem>()
+    }
+
+    private fun <T> MutableLiveData<T>.postOnUi(value: T) {
```

Comment:
It is my understanding is that setValue() must be called from the main thread whereas postValue() is used when you want to set a message from a background thread. In this case I want to ensure that the change is posted on the main thread and observers are notified accordingly.

Reply:
Hmm, interesting. My understanding was that `setValue` needs to be called on the main thread as it's a blocking operation requiring UI thread. 
Whereas `postValue` doesn't need to be performed on the main thread as it's a non-blocking operation. 
However, even `postValue` eventually sets the value on the main thread, since LiveData are observed from the UI and we can't change the UI from a background thread.
So I'm still not sure  if our custom `postOnUi` actually does anything special or it's just a custom implementation of `postValue`. Am I missing something?

Reply:
Actually I didn't have the need for this code any longer, it has been removed.